src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -16,57 +16,118 @@
 import io.stargate.sgv2.jsonapi.service.shredding.model.WritableShreddedDocument;
 import io.stargate.sgv2.jsonapi.service.updater.DocumentUpdater;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Supplier;
-import java.util.stream.Stream;
 
+/**
+ * This operation method is used for 3 commands findOneAndUpdate, updateOne and updateMany
+ *
+ * @param commandContext
+ * @param readOperation
+ * @param documentUpdater
+ * @param returnDocumentInResponse
+ * @param returnUpdatedDocument
+ * @param upsert
+ * @param shredder
+ * @param updateLimit
+ */
 public record ReadAndUpdateOperation(
     CommandContext commandContext,
     ReadOperation readOperation,
     DocumentUpdater documentUpdater,
     boolean returnDocumentInResponse,
     boolean returnUpdatedDocument,
     boolean upsert,
-    Shredder shredder)
+    Shredder shredder,
+    int updateLimit)
     implements ModifyOperation {
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
-    Uni<ReadOperation.FindResponse> docsToUpate = readOperation().getDocuments(queryExecutor, null);
+    final AtomicBoolean moreDataFlag = new AtomicBoolean(false);
+    final Multi<ReadOperation.FindResponse> findResponses =
+        Multi.createBy()
+            .repeating()
+            .uni(
+                () -> new AtomicReference<String>(null),
+                stateRef -> {
+                  Uni<ReadOperation.FindResponse> docsToUpdate =
+                      readOperation().getDocuments(queryExecutor, stateRef.get());
+                  return docsToUpdate
+                      .onItem()
+                      .invoke(findResponse -> stateRef.set(findResponse.pagingState()));
+                })
+            .whilst(findResponse -> findResponse.pagingState() != null);
+    final AtomicInteger matchedCount = new AtomicInteger(0);
+    final AtomicInteger modifiedCount = new AtomicInteger(0);
+
     final Uni<List<UpdatedDocument>> updatedDocuments =
-        docsToUpate
+        findResponses
             .onItem()
             .transformToMulti(
                 findResponse -> {
-                  if (findResponse.docs().isEmpty()) {
-                    if (upsert) {
-                      return Multi.createFrom().item(readOperation().getEmptyDocuments());
+                  final List<ReadDocument> docs = findResponse.docs();
```

Comment:
:warning: Same issue as for the `deleteMany`.

Reply:
I have explained this in deleteMany PR. Once it's resolved will resolve here.